src/map.rs

```diff
@@ -153,11 +153,18 @@ impl Map {
         let mut tilesets = Vec::new();
 
         parse_tag!(parser, "map", {
-            "tileset" => |attrs| {
-                let res = Tileset::parse_xml_in_map(parser, attrs, map_path)?;
+            "tileset" => |attrs: Vec<OwnedAttribute>| {
+                let res = Tileset::parse_xml_in_map(parser, &attrs, map_path,  reader, cache)?;
                 match res.result_type {
                     EmbeddedParseResultType::ExternalReference { tileset_path } => {
-                        let tileset = cache.get_or_try_insert_tileset_with(tileset_path.clone(), || crate::parse::xml::parse_tileset(&tileset_path, reader))?;
+                        let tileset = if let Some(ts) = cache.get_tileset(&tileset_path) {
+                            ts
+                        } else {
+                            let tileset = Arc::new(crate::parse::xml::parse_tileset(&tileset_path,  reader, cache)?);
+                            cache.insert_tileset(tileset_path.clone(), tileset.clone());
+                            tileset
+                        };
```

Comment:
Could we, instead of this construct, have passed the `cache` as an argument to the lambda used by `get_or_try_insert_tileset_with`? Or is that not preferable?

Reply:
Can't do because of borrow rules, get_or_try_insert_with uses &mut cache and passing &cache into the closure wouldn't be possible.

Reply:
I find this quite unexpected. Surely a mutable reference can be turned into a non-mutable reference? Note that I'm asking about passing it in as a parameter, not as a captured value.

Reply:
Oh I see what you say now... Yeah, that would be possible. We can try that.

Reply:
Actually never mind, I think it'll still not work, since `parse_tileset` actually need a mutable reference, and we probably can't create another one with the vacant hash map entry around. And if we work around that, the advantage of using `tilesets.entry` is gone anyway.